### PR TITLE
Examples, and help files.

### PR DESCRIPTION
**Pull request by riddlez666**
I reorganized and added a folder section called example-configs for not only me to merge stuff into but for others to share there setups, which in turn will help everyone in the end. Moved the misc folder into the example-configs as it seemed fitting.

### CODE REVIEW

**Comment by riddlez666** [comment]
@ellaismer merge please

**Comment by ellaismer** [comment]
I'll get back on this really soon. Ultimately I'm still troubled by the duplicate configs. The example config consists of many duplicate sections, and it will be a huge maintenance burden if we update anything -- we need to update all 6 example configs for any change.

So I'm thinking about getting the duplication problem fixed in the config module before we merge example configs.

Also some minor structural issue. Do you mind to keep the existing `misc` folder at top-level? That seems to be more general. So we have

```
./misc
./misc/upstart
./misc/example-configs
```

**Comment by riddlez666** [comment]
I think the folder structure fits as the upstart.sh is a example config like the others. And as a new user setting up a pool i would navigate to example-configs before i would to misc. Yes the duplication is a issue but it teaches the user to separate the threads, which is useful if there pool grows.

**Comment by ellaismer** [comment]
Okay then. Let's fix the duplication problem later.

Still `misc` means "miscellaneous" and is a more general term. Given that `upstart` is also config files. How about we just move the current `example-configs` to `misc`? So we have:

```
./misc
./misc/example-configs
./misc/example-configs/upstart
```

If no problem, I'm going to change this and merge this PR.

File: example-configs/accounts/ellauser [hunk 1]
```diff
@@ -0,0 +1 @@
+0x0035e462b6026C3b26A77B0Ad917a0be0C9f419F
```
Comment (ellaismer) [inline example-configs/accounts/ellauser]: Also suggest to change this to `0x0000..0000` to force user to change it.

File: example-configs/api.json [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+		"endpoint": "127.0.0.1:6379",
+		"poolSize": 10,
+		"database": 0,
+		"password": ""
+	},
+	"unlocker": {
+		"enabled": false,
+		"poolFee": 1.0,
+		"poolFeeAddress": "",
+		"donate": false,
```
Comment (ellaismer) [inline example-configs/api.json:74]: Also suggest to change this to `true` as it helps the community.

**Comment by riddlez666** [comment]
Yes those will do. Merge away.

**Comment by riddlez666** [comment]
btw with the threads separated it would need to be in unlocker.json not api.json

**Review by ellaismer — APPROVED** [review]

**Pull request merged**